app/models/project_group.rb

```diff
@@ -54,6 +54,21 @@ def summary_or_none
     summary.blank? ? "[#{I18n.t("none")}]" : summary.to_s
   end
 
+  def scheduled_start_date
+    return @scheduled_start_date if defined?(@scheduled_start_date)
+    @scheduled_start_date = reload.children.map(&:scheduled_start_date).compact.min
+  end
+
+  def scheduled_end_date
+    return @scheduled_end_date if defined?(@scheduled_end_date)
+    @scheduled_end_date = reload.children.map(&:scheduled_end_date).compact.max
+  end
+
+  def scheduled_duration_days
+    return @scheduled_duration_days if defined?(@scheduled_duration_days)
+    @scheduled_duration_days = scheduled_end_date - scheduled_start_date if scheduled_end_date && scheduled_start_date
```

Comment:
Shouldn't we set it to nil here and memoize that if start or end isn't set? So use a ternary?